thunder/executors/torchex.py

```diff
@@ -1883,6 +1887,21 @@ def _update_bucket_view_prim_impl(
         views[index_of_dst_view].copy_(tensor)
         return tensor
 
+    # TODO(crcrpar): Update to return None instead
+    def _stash_grad_for_fsdp_prim_impl(
+        grad: torch.Tensor,
+        param_fqn: str,
+        compile_data: CompileData,
+    ) -> None:
+        grad_name = "_thunder_fsdp_unsharded_grad"
+        param = compile_data.fn.get_parameter(param_fqn)
+        if torch.is_tensor(unsharded_grad := getattr(param, grad_name, None)):
+            unsharded_grad += grad
+        else:
+            setattr(param, grad_name, grad)
```

Comment:
This `_stash_grad_for_fsdp_prim_impl` is great, I think we should do this accumulation in the general case as well. It gives us more control.
In https://github.com/Lightning-AI/lightning-thunder/pull/214 I'm experimenting with creating symbol that would be called in the forward trace and do `ThunderFunction.apply`.